### PR TITLE
Fix large files upload

### LINKED ISSUE #52: File upload doesn't work for large files
I'm running "#type -> file upload" test from Element.js suite via browserstack. It works when I use default upload.txt file or any file under ~60Kb. If I use any file bigger than ~60Kb I receive the following exception:

```
oversubscribed dynamic bit lengths tree
```

Screenshot:

screen shot 2015-08-28 at 10 49 08 am

This exception goes from java.util.zip.ZipException, which is probably returned from Selenium server. It means that zip file is invalid / corrupted (http://stackoverflow.com/questions/3049448/java-util-zip-zipexception-oversubscribed-dynamic-bit-lengths-tree)

I think it can be related to adm-zip library implementation. 

**Steps to reproduce**: 
1) Add any file bigger than 60Kb to /tests/functional/data/ folder
2) Edit "#type -> file upload" test from Element.js suite: replace `upload.txt` with the filename from 1), update expected file size
3) Run "#type -> file upload" test from Element.js suite

**Expected output**: successful test

**Actual output:**

```
[reduced...]nPee767/9+hdQSwECCgAKAAAACAC8VQVHxVtD+n85AAAA5gAAEwAAAAAAAAAAALYBAAAAAAAAMTUwX3RyYW5zYWN0aW9uLnhsc1BLBQYAAAAAAQABAEEAAACwOQAAAAA="}] oversubscribed dynamic bit lengths tree
  at Session._post  </Users/sap1ens/Dev/leadfoot/Session.js:59:30>
  at </Users/sap1ens/Dev/leadfoot/Session.js:1687:16>
  at new Promise  <node_modules/dojo/Promise.ts:411:3>
  at Session._uploadFile  </Users/sap1ens/Dev/leadfoot/Session.js:1681:9>
  at Element.type  </Users/sap1ens/Dev/leadfoot/Element.js:200:24>
  at <tests/functional/Element.js:80:21>
  at <node_modules/dojo/Promise.ts:393:15>
  at runCallbacks  <node_modules/dojo/Promise.ts:11:11>
  at <node_modules/dojo/Promise.ts:317:4>
  at run  <node_modules/dojo/Promise.ts:237:7>
```

And you can also check the screenshot from Browserstack side.

**Additional details**:
OS X 10.10.4
Leadfoot: master branch

### PR DESCRIPTION
**Pull request by sap1ens**
Closes #52 

CLA is signed

### CODE REVIEW

**Comment by trevoro** [comment]
Looks good to me.
Left a note about `zip = null`.

File: Session.js [hunk 1]
```diff
@@ -1679,9 +1681,11 @@ Session.prototype = {
...
 		return new Promise(function (resolve) {
-			var zip = new AdmZip();
-			zip.addLocalFile(filename);
-			var data = zip.toBuffer().toString('base64');
+			var content = fs.readFileSync(filename);
+
+			var zip = new JSZip();
+			zip.file(path.basename(filename), content);
+			var data = zip.generate({type: 'base64'});
 			zip = null;
```
Comment (trevoro) [inline Session.js]: What's the intention here?

Comment (sap1ens) [inline Session.js]: Keeping it from the old codebase. As I understand it helps with GC and avoiding memory leaks.

**Comment by csnover** [comment]
Sorry for the delay, the biggest concern I have with this change is that adm-zip is already used as a dependency in a sibling project (Dig Dug, as a sub-dependency of the decompress package) so this leaves us using two different ZIP processing libraries. I know that having 9000 dependencies is trendy among npm publishers but I try to avoid having so many dependencies since it makes downloading a chore. Do you know if this issue is resolved in a more recent release of adm-zip or is it just terminally broken no matter what?

**Comment by sap1ens** [comment]
Hey, no worries. I see your point.

I've tried to play with the latest version of adm-zip locally, but I gave up too early, it looked completely broken. And based on number and types of the issues here https://github.com/cthackers/adm-zip/issues I do not recommend to use adm-zip at all.

**Comment by csnover** [comment]
I don’t blame you for not recommending it. Looks like the latest version of decompress is using something totally different anyway and has a million extra dependencies that don’t exist in the older version, so it’s probably healthier for everyone to just switch the library used here, like this PR does. :(

**Pull request closed without merging**